### PR TITLE
Support parsing unary plus (fixes #1720)

### LINKED ISSUE #1720: sympify syntax error: unary plus operator
```python
symengine.sympify('+x')
```
gives
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "symengine_wrapper.pyx", line 500, in symengine.lib.symengine_wrapper.sympify
RuntimeError: syntax error
```
(symengine-0.6.1 and Python 3.9.1)

What I expected is returning a symbol as sympy does (`sympy.sympify('+x')`).

### PR DESCRIPTION
**Pull request by rikardn**
Add  support for unary plus in grammar.

### CODE REVIEW

**Comment by rikardn** [comment]
This didn't go well. I am thinking it might be my version of bison that is generating buggy code. Has anyone seen anything like this?

File: symengine/parser/parser.yy [hunk 1]
```diff
@@ -180,6 +180,9 @@ expr:
 |
         '-' expr %prec UMINUS
         { $$ = neg($2); }
+|
+        '+' expr %prec UMINUS
```
Comment (lkeegan) [inline symengine/parser/parser.yy:184]: doesn't change the results, but maybe for consistency with the other rules? (along with adding a `%right UPLUS` below the existing `%right UMINUS` on line 89)
```suggestion
        '+' expr %prec UPLUS
```

**Review by lkeegan — APPROVED** [review]
lgtm!

I fixed the compiler warning that was causing the debug CI builds to fail, now it looks like tokenizer.cpp is causing clang_format problems, but as you didn't change tokenizer.re maybe the easiest thing is just to revert the changes to this file?

File: symengine/parser/parser.yy (hunk 1 above)
Comment (rikardn) [reply symengine/parser/parser.yy]: Ok I have fixed this

**Comment by rikardn** [comment]
Thanks @lkeegan for the help! I opted to keep the newly generated tokenizer and fix the code formatting. There might be some benefit to have it being generated by a more recent version of re2c.

**Pull request merged**